### DISCUSSION TITLE
Recommendations for splitting a large API into multiple handlers?

### DISCUSSION THREAD

**Discussion by johnrutherford** [ogen-go/ogen#1181] [General]
If you have a large API, implementing all endpoints in single large `Handler` will be difficult to maintain. The GitHub API is an extreme [example](https://github.com/ogen-go/ogen/blob/main/examples/ex_github/oas_server_gen.go#L10). 

Typically you would want to split up your implementation by version, resource, etc. How can you do this with `ogen`? It looks like [contiamo/openapi-generator-go](https://github.com/contiamo/openapi-generator-go) uses `x-handler-group` to group operations into separate handlers.

**Comment by samstarling** [comment]
I also had the same question! Did you ever find a good answer to this>

**Comment by jaffyyyy** [comment]
up

**Comment by corysullivan** [comment]
You can use [Operation Groups](https://ogen.dev/docs/spec/extensions#operation-groups)

**Reply by japusoy** [reply]
Thanks!
